app/src/main/java/com/arctouch/codechallenge/ui/detail/DetailsFragment.kt

```diff
@@ -0,0 +1,77 @@
+package com.arctouch.codechallenge.ui.detail
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.Logger
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder
+import com.arctouch.codechallenge.util.convertToPtBrDateFormat
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.details_fragment.*
+import org.koin.android.viewmodel.ext.android.sharedViewModel
+
+class DetailsFragment : Fragment() {
+    private var TAG = Logger.tag
+    private val viewModel by sharedViewModel<DetailsViewModel>()
+
+    companion object {
+        fun newInstance() = DetailsFragment()
+    }
+
+    private val movieImageUrlBuilder = MovieImageUrlBuilder()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View {
+        return inflater.inflate(R.layout.details_fragment, container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        loadData()
+
+    }
+
+    @SuppressLint("CheckResult")
```

Comment:
Why is there a warning in the first place and how could it affect your code? Is suppressing the warning without fixing it the best option?

Reply:
Because I inserted the null cehck, nothing will happen, because I did not implement the error flow in the Observable. ProgressBar will be running forever.

Reply:
This suppressor serves only to not show an IDE alert, in which case the ViewModel's Observable return is not being used. We could have brought the answer in another method, but I believe that this series of calls does not look good.